pkg/resources/secret_with_basic_authentication.go

```diff
@@ -0,0 +1,163 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var secretBasicAuthenticationSchema = func() map[string]*schema.Schema {
+	secretBasicAuthentication := map[string]*schema.Schema{
+		"username": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the username value to store in the secret when setting the TYPE value to PASSWORD.",
+		},
+		"password": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the password value to store in the secret when setting the TYPE value to PASSWORD.",
```

Comment:
Use externalChangesNotDetectedFieldDescription for the description.

Reply:
changed to use `externalChangesNotDetectedFieldDescription`

Reply:
Should it be used only in this filed or in other fields as well? Or for fields with `Sensitive: true` like this one or secret_string?

Reply:
In general, this applies to all the fields that are not available to read from Snowflake, particularly sensitive fields, like passwords and keys, are not available.